src/renderers/shared/fiber/ReactChildFiber.js

```diff
@@ -428,32 +435,37 @@ function ChildReconciler(shouldClone, shouldTrackSideEffects) {
           const created = createFiberFromElement(newChild, priority);
           created.ref = coerceRef(null, newChild);
           created.return = returnFiber;
+          created.contextTag = returnFiber.contextTag;
           return created;
         }
 
         case REACT_COROUTINE_TYPE: {
           const created = createFiberFromCoroutine(newChild, priority);
           created.return = returnFiber;
+          created.contextTag = returnFiber.contextTag;
           return created;
         }
 
         case REACT_YIELD_TYPE: {
           const created = createFiberFromYield(newChild, priority);
           created.type = newChild.value;
           created.return = returnFiber;
+          created.contextTag = returnFiber.contextTag;
           return created;
         }
 
         case REACT_PORTAL_TYPE: {
           const created = createFiberFromPortal(newChild, priority);
           created.return = returnFiber;
+          created.contextTag = returnFiber.contextTag;
           return created;
         }
       }
 
       if (isArray(newChild) || getIteratorFn(newChild)) {
         const created = createFiberFromFragment(newChild, priority);
         created.return = returnFiber;
+        created.contextTag = returnFiber.contextTag;
```

Comment:
It probably makes sense to pass more arguments to the createFiber call though so that we can initialize everything as the correct value from the start.